test/AbstractMemoryTest.cpp

```diff
@@ -0,0 +1,209 @@
+// Copyright 2023, University of Freiburg,
+// Chair of Algorithms and Data Structures.
+// Author: Andre Schlegel (July of 2023,
+// schlegea@informatik.uni-freiburg.de)
+
+#include <absl/strings/str_cat.h>
+#include <gtest/gtest.h>
+
+#include <algorithm>
+#include <cstdint>
+#include <ranges>
+#include <vector>
+
+#include "util/AbstractMemory/Memory.h"
+
+// Importing all the literals.
+using ad_utility::operator""_Byte;
+using ad_utility::operator""_KB;
+using ad_utility::operator""_MB;
+using ad_utility::operator""_GB;
+using ad_utility::operator""_TB;
+using ad_utility::operator""_PB;
+
+TEST(AbstractMemory, UserDefinedLiterals) {
+  // Normal bytes.
+  ASSERT_EQ(50uL, 50_Byte);
+
+  // Kilobytes.
+  ASSERT_EQ(2048uL, 2_KB);    // Whole number.
+  ASSERT_EQ(1536uL, 1.5_KB);  // Floating point without rounding.
+  ASSERT_EQ(1332uL, 1.3_KB);  // Floating point with rounding.
```

Comment:
I think we definitely shouldn't have implicit conversion or comparison to normal integers, that defeats the purpose of a strong type.
Either you have to be forced to call `.bytes()` or at least make the operator explicit s.t. we have to call `static_cast`, although I prefer the first stronger version.

Reply:
Do I understand correctly, that you don't want the user defined literals to create a `size_t`, but instead to create an instance of `Memory`?

Reply:
Exactly!

Reply:
Done, see [f5f9f83](https://github.com/ad-freiburg/qlever/pull/1032/commits/f5f9f839efb6fcc21d1e4310d3b974e985ce8e91).